gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/MDGExtension.groovy

```diff
@@ -2,135 +2,141 @@ package org.groovymc.modsdotgroovy.gradle
-            else if (project.plugins.findPlugin('net.neoforged.gradle.userdev')) return Map.of(Platform.NEOFORGE, projectPath)
-            else if (project.plugins.findPlugin('fabric-loom')) return Map.of(Platform.FABRIC, projectPath)
-            else if (project.plugins.findPlugin('org.quiltmc.loom')) return Map.of(Platform.QUILT, projectPath)
-        } else {
-            // subprojects, so account for the possibility of multiple platforms and guess based on the names of the subprojects
-            // (we can't check the subprojects' plugins from the root project because they haven't been applied yet)
-            final Map<Platform, List<String>> platforms = [:]
-            for (Project subproject in project.subprojects) {
-                switch (subproject.name) {
-                    case 'forge':
-                        if (platforms.containsKey(Platform.FORGE)) platforms[Platform.FORGE] << subproject.path
-                        else platforms[Platform.FORGE] = [subproject.path]
-                        break
-                    case 'neoforge':
-                        if (platforms.containsKey(Platform.NEOFORGE)) platforms[Platform.NEOFORGE] << subproject.path
-                        else platforms[Platform.NEOFORGE] = [subproject.path]
-                        break
-                    case 'fabric':
-                        if (platforms.containsKey(Platform.FABRIC)) platforms[Platform.FABRIC] << subproject.path
-                        else platforms[Platform.FABRIC] = [subproject.path]
-                        break
-                    case 'quilt':
-                        if (platforms.containsKey(Platform.QUILT)) platforms[Platform.QUILT] << subproject.path
-                        else platforms[Platform.QUILT] = [subproject.path]
-                        break
-                    case 'spigot':
-                        if (platforms.containsKey(Platform.SPIGOT)) platforms[Platform.SPIGOT] << subproject.path
-                        else platforms[Platform.SPIGOT] = [subproject.path]
-                        break
-                }
-            }
-
-            return platforms.collectEntries { Platform platform, List<String> paths -> [platform, paths.toArray(String[]::new)] }
+        void from(String projectPath, String sourceSetName) {
+            var configurationName = forSourceSetName(sourceSetName, EXPOSE_SOURCE_SET)
+            var consumingConfiguration = project.configurations.detachedConfiguration(
+                    project.dependencies.project(path: projectPath, configuration: configurationName)
+            )
+            consumingConfiguration.canBeResolved = true
+            consumingConfiguration.canBeConsumed = false
+            getModsDotGroovyFile().set(consumingConfiguration)
+            multiplatform.set(true)
         }
+    }
+
+    private static final String EXPOSE_SOURCE_SET = 'shareModsDotGroovy'
+    private static final String DEFAULT_MDG = "mods.groovy"
```

Comment:
Private static finals at the top please, took me a min finding this because I'm used to it either being next to the first usage or at the top.